### PR TITLE
Improve rustdocs on addresses module

### PR DESCRIPTION
**Pull request by tcharding**
These docs are stale, update them.

- Mention segwit and legacy
- Improve example code to show feature gating
- Fix headings to be as typical

### CODE REVIEW

File: bitcoin/src/address/mod.rs [hunk 1]
```diff
@@ -2,25 +2,28 @@
...
 //!
-//! // Generate pay-to-pubkey-hash address.
+//! // Generate a mainnet pay-to-pubkey-hash address.
 //! let address = Address::p2pkh(&public_key, Network::Bitcoin);
-//! # }
+//! }
 //! ```
 //!
-//! # Note: creating a new address requires the rand-std feature flag
+//! Note: creating a new address requires the rand-std feature flag.
```
Comment (apoelstra) [inline bitcoin/src/address/mod.rs:26]: In f5a774dbd96140ae36435d9591566c9dcd3220f6:

We should say "creating a new address in this way". There are lots of ways to create addresses that do not require a RNG.

Comment (tcharding) [reply bitcoin/src/address/mod.rs]: Do we even need this comment now that this PR makes the feature gating on the example visible?

File: bitcoin/src/address/mod.rs [hunk 2]
```diff
@@ -2,25 +2,28 @@
 
 //! Bitcoin addresses.
 //!
-//! Support for ordinary base58 Bitcoin addresses and private keys.
+//! Support for segwit and legacy addresses (bech32 and base58 respectively).
```
Comment (storopoli) [inline bitcoin/src/address/mod.rs:5]:
```suggestion
//! Support for SegWit and legacy addresses (bech32 and base58 respectively).
```

File: bitcoin/src/address/mod.rs [hunk 3]
```diff
@@ -2,25 +2,28 @@
...
 //!
-//! # Example: creating a new address from a randomly-generated key pair
+//! # Examples
+//!
+//! ### Creating a new address from a randomly-generated key pair.
```
Comment (storopoli) [inline bitcoin/src/address/mod.rs:9]: Why you are going from H1 to H3?
```suggestion
//! # Examples
//!
//! ## Creating a new address from a randomly-generated key pair.
```

File: bitcoin/src/address/mod.rs [hunk 4]
```diff
@@ -2,25 +2,28 @@
...
 //!
-//! // Generate pay-to-pubkey-hash address.
+//! // Generate a mainnet pay-to-pubkey-hash address.
 //! let address = Address::p2pkh(&public_key, Network::Bitcoin);
-//! # }
+//! }
 //! ```
 //!
-//! # Note: creating a new address requires the rand-std feature flag
+//! Note: creating a new address in this way requires the rand-std feature flag.
```
Comment (storopoli) [inline bitcoin/src/address/mod.rs:26]:
```suggestion
//! Note: creating a new address in this way requires the `rand-std` feature flag.
```

**Review by storopoli — COMMENTED** [review]
Some suggestions.

File: bitcoin/src/address/mod.rs (hunk 3 above)
Comment (tcharding) [reply bitcoin/src/address/mod.rs:9]: Because H1 is rendered the same as H2 in rustdocs

Comment (tcharding) [reply bitcoin/src/address/mod.rs:9]: Unless I'm mistaken

**Comment by apoelstra** [comment]
LGTM -- will ACK but hold off on merging until @storopoli suggestions are responded to

File: bitcoin/src/address/mod.rs (hunk 3 above)
Comment (storopoli) [reply bitcoin/src/address/mod.rs:9]: Ok, just warned because it seemed like an error. But if it was intentional I don't mind.

**Comment by storopoli** [comment]
> LGTM -- will ACK but hold off on merging until @storopoli suggestions are responded to

I think just the `rand-std` using backticks for code formatting is my only blocker.

File: bitcoin/src/address/mod.rs (hunk 2 above)
Comment (tcharding) [reply bitcoin/src/address/mod.rs:5]: Leaving this for #3770

File: bitcoin/src/address/mod.rs (hunk 3 above)
Comment (tcharding) [reply bitcoin/src/address/mod.rs:9]: Cool, thanks man.

Comment (tcharding) [reply bitcoin/src/address/mod.rs:9]: I rendered the docs and checked the HTML. I usually do that for rustdocs changes but I appreciate you noticing things. Thanks

File: bitcoin/src/address/mod.rs (hunk 1 above)
Comment (tcharding) [reply bitcoin/src/address/mod.rs]: Didn't get a response here so I just removed the note all together.

**Comment by tcharding** [comment]
The `rand-std` thing is fixed because the whole line is removed.

**Review by storopoli — APPROVED** [review]
ACK f85456a726a201b26eac58d22eb0eb6a58ad411b

**Review by apoelstra — APPROVED** [review]
ACK e3761d240f28b0a0b51326f40ab986675e7810a1; successfully ran local tests

**Review by apoelstra — APPROVED** [review]
ACK f85456a726a201b26eac58d22eb0eb6a58ad411b; successfully ran local tests

**Pull request merged**